sdcm/nemesis.py

```diff
@@ -2800,6 +2802,57 @@ def _verify_cdc_feature_status(self, keyspace: str, table: str, cdc_settings: di
             assert actual_cdc_settings == cdc_settings, \
                 f"CDC extension settings are differs. Current: {actual_cdc_settings} expected: {cdc_settings}"
 
+    def _add_new_node_in_new_dc(self) -> BaseNode:
```

Comment:
This doesn't follow the procedure we have of "Adding new DC to existing cluster".
https://docs.scylladb.com/operating-scylla/procedures/cluster-management/add-dc-to-existing-dc/

In this doc, it explains exactly  how to add the new DC with firstly clean nodes that probabbly shouldn't bootstrap.
There are changes needed in RF of authenticator (maybe we already  have them by default), but what worries me the most is the change of existing "app" (in our case the load) from QUORUM to LOCAL_QUORUM.

What I suggest is as follow:
For now, don't change this code in nemesis.
Add a new test in SCT (maybe under new module called "cluster_management_procedures_test.py).
The test should cover the procedure exactly as it written, but for now don't change the stress (that you will start on the basic cluster when it's ready before the procudere) from QUORUM to LOCAL_QUORUM. 
Then, when we will understand better how it works and affect the cluster, we can decide if to adopt the same approach for the nemesis, or we will keep the nemesis as is (we will need to consult Shlomi before that).

Reply:
I thought we don't need node cleanup as we create it in a new instance and set up as new dc before first startup.

I'll add step regarding increase rf for system_auth to match number of nodes before adding new dc.

We don't need steps for restarting nodes in existing DC as I'm not changing any of the settings in any existing node. Here's some exception from the flow - I'm not adding new seed nodes - shall I add this part? In that case restart of all nodes will be required.

Regarding client QUORUM to LOCAL_QUORUM change - currently we don't change replication strategy for existing keyspaces (c-s load keyspaces) - in that case I'm not sure if we should care. Shall I make user keyspaces to use new DC?

Reply:
Regarding all these steps, well I don't know.
At the moment, you invented a procedure, which might work, might be relevant for our case, or whatever.

I want to follow the exact procedure we have in the documentation, so we test it and if we have issues it's not because we did something that is unsupported.

I don't mind that this nemesis will add 2 new nodes in that new DC and then remove them.

Regarding the QUORUM and LOCAL_QUORUM, I have no idea how it behaves, I don't know why we have this comment in the documentation, and I don't know why it's relevant if the NetworkTopology hasn't changed yet to include the new DC.
While testing, we can open an issue for the documentation.

Reply:
I've tried to follow as closely as needed - some steps in my opinion are redundant/too costly.
Exemptions from the procedure are:
1. Didn't modify snitch in existing DC as is already set to Ec2MultiRegionSnitch - so no reconfigure&restart existing DC nodes is required
2. didn't clean new node data (I was thinking that is not required if created from scratch)
3. not changing existing user keyspaces to use new DC (that's why in my opinion we don't need to change client's CL) 
4. not setting seed nodes in new DC (so there's no need to reconfigure&restart all the nodes)
5. new DC is in the same region as original DC (which is not exemption, but rather hack)

@roydahan if you consider any of this as an error - let me know, I'll adapt the test flow to include it.

Reply:
Well, maybe for this nemesis it's ok.
We will need to add another task and check the procedure of adding a new DC to an existing workload like the procedure and see if it makes any different.